base57.go

```diff
@@ -45,7 +45,8 @@ func (b *base57) numToString(number *big.Int, padToLen int) string {
 		digit *big.Int
 	)
 
-	for number.Uint64() > 0 {
+	zero := new(big.Int)
+	for number.Cmp(zero) > 0 {
```

Comment:
Could maybe get away with using [`big.Int.BitLen()`](https://golang.org/pkg/math/big/#Int.BitLen) as well 🤔 Not sure which is clearer.

> BitLen returns the length of the absolute value of x in bits. The bit length of 0 is 0.

```suggestion
	for number.BitLen() > 0 {
```

Reply:
I think either would work just fine, but I don't have any sense of which would be more efficient or anything (I hadn't heard of `BitLen()` before now).  I chose `Cmp()` because I was familiar with it and its intent was clear to me.

Reply:
Agreed. Using `Cmd()` conveys the intent better.